Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -869,6 +881,40 @@ private string GetZipDeployPathInfo()
             return pathmsg;
         }
 
+        private void BackupErrorZip(ArtifactDeploymentInfo zipDeploymentInfo)
+        {
+            try
+            {
+                // File created in D:\home\data\SitePackages for current deployment
+                var filePath = Path.Combine(_environment.SitePackagesPath, zipDeploymentInfo.ArtifactFileName);
+
+                // Store only if zip file size < 100 MB
+                long fileBytes = new FileInfo(filePath).Length;
+                float fileMB = (float)fileBytes / (1024 * 1024);
+                if (fileMB < 100)
+                {
+                    // Make sure D:\home\LogFiles\kudu\deployment\BackupErrorZip exists
+                    var errorDirPath = Path.Combine(_environment.DeploymentTracePath, "BackupErrorZip");
+                    FileSystemHelpers.EnsureDirectory(errorDirPath);
+
+                    // Store only 1 zip file with error
+                    IEnumerable<string> zipFiles = FileSystemHelpers.GetFiles(errorDirPath, "*.zip");
+                    if (zipFiles.Count() == 0)
+                    {
+                        var errorFilePath = Path.Combine(errorDirPath, zipDeploymentInfo.ArtifactFileName);
+                        using (_tracer.Step("{0}: Saving zip file with error to {1} of size {2} MB.", nameof(BackupErrorZip), errorFilePath, fileMB))
```

Comment:
make sure this show up on Kudu kusto table as we would need it for chasing this.

Reply:
Yes, I have tested it. Logs are created in Kusto.